controllers/front/Payment.php

```diff
@@ -102,10 +111,10 @@ public function postProcess()
 
         if (!\Validate::isLoadedObject($customer)) {
             if (empty($_REQUEST['isAjax'])) {
-                \Tools::redirect('index.php?controller=order&step=1');
+                \Tools::redirect($this->context->link->getPageLink('order', $this->ssl) . '?step=1');
             } else {
                 $this->ajaxRender($this->helperData->buildControllerResponseJson('redirect',
-                    ['redirectUrl' => 'index.php?controller=order&step=1']));
+                    ['redirectUrl' => $this->context->link->getPageLink('order', $this->ssl) . '?step=1']));
```

Comment:
Do we need to add the question mark here?

Reply:
we can pass the request as a parameter and then the function will decide how to concatenate.